lens/private/syntax/syntax.rkt

```diff
@@ -53,14 +53,14 @@
       [(a ...) (ormap (target-stx target-id) (syntax->list #'(a ...)))]
       [a (and (bound-identifier=? target-id #'a) #'a)]))
   
-  (define template->pattern
-    (syntax-parser #:literals (_)
-      [(a ...) #`(#,@(stx-map template->pattern #'(a ...)))]
-      [_ (generate-temporary)]
+  (define (template->pattern stx)
+    (syntax-parse stx
+      [(a ...) (datum->syntax stx (stx-map template->pattern #'(a ...)) stx stx)]
```

Comment:
Gotcha. I'm fine with this change then, I just initially wasn't sure what it was for.